apiExamples/nestedSelect.js

```diff
@@ -0,0 +1,40 @@
+export function openDialog() {
+  // JavaScript for the overflow: visible dialog
+  let visOverflowDialog = document.getElementById('visibleOverflowDialog');
+  let visibleOverflowButton = document.getElementById('overflowVisibleButton');
+  let closeVisButtonElem = document.getElementById('closeVisButton');
+
+  visibleOverflowButton.addEventListener('click', () => {
+    visOverflowDialog.setAttribute('open', 'true');
+  });
+
+  closeVisButtonElem.addEventListener('click', () => {
+    visOverflowDialog.removeAttribute('open');
+  });
+
+  // JavaScript for the overflow: hidden dialog
+  let hidOverflowDialog = document.getElementById('hiddenOverflowDialog');
+  let hiddenOverflowButton = document.getElementById('overflowHiddenButton');
+  let closeHidButtonElem = document.getElementById('closeHidButton');
+
+  hiddenOverflowButton.addEventListener('click', () => {
+    hidOverflowDialog.setAttribute('open', 'true');
+  });
+
+  closeHidButtonElem.addEventListener('click', () => {
+    hidOverflowDialog.removeAttribute('open');
+  });
+
+  // JavaScript for the overflow: visible with overflowing content dialog
+  let overflowContentDialog = document.getElementById('overflowContentDialog');
+  let overflowContentButton = document.getElementById('overflowContentButton');
+  let closeOverflowButtonElem = document.getElementById('closeOverflowButton');
+
+  overflowContentButton.addEventListener('click', () => {
+    overflowContentDialog.setAttribute('open', 'true');
+  });
+
+  closeOverflowButtonElem.addEventListener('click', () => {
+    overflowContentDialog.removeAttribute('open');
+  });
+}
```

Comment:
This is a LOT of repeated code. Please take a look at the [demo code](https://auro.alaskaair.com/components/auro/interruption/dialog#gsc.tab=0:~:text=of%20closing%20it-,The%20setup,-Triggering%20the%20dialog) for auro-dialog. 

Let's reduce this demo code as not to overwhelm any user of this feature.

Reply:
I looked at the demo code for auro-dialog and while it is very simplified and ideal for this demo, I don't think we can use that logic without some kind of hacking, due to new exampleJS system Jason has been working on.

If you look in api.js, you'll see that we have to call the function immediately to initialize the example. I would have to pass in one of the two dialogs immediately and run the code within my `openDialog()` function, which is not the behavior we want. I've removed the third dialog to make the code less heavy, but I think the best options are:

1. Do not use the exampleJS system and import the Javascript file for this example directly into `api.html` and have a clean JS file for the example, like we do in auro-dialog.
2. Slim down the code in `openDialog()` and have a helper function that behaves like the demo code from auro-dialog.
3. Leave the JS code as is.